core/server/data/migrations/versions/2.0/2-update-posts.js

```diff
@@ -2,32 +2,66 @@ const _ = require('lodash'),
     Promise = require('bluebird'),
     common = require('../../../../lib/common'),
     models = require('../../../../models'),
-    message1 = 'Updating post data (comment_id)',
-    message2 = 'Updated post data (comment_id)',
-    message3 = 'Rollback: Keep correct comment_id values in amp column.';
+    converters = require('../../../../lib/mobiledoc/converters'),
+    message1 = 'Updating posts: apply new editor format and set comment_id field.',
+    message2 = 'Updated posts: apply new editor format and set comment_id field.',
+    message3 = 'Rollback: Updating posts: use old editor format',
+    message4 = 'Rollback: Updated posts: use old editor format';
 
 module.exports.config = {
     transaction: true
 };
 
+let mobiledocIsCompatibleWithV1 = function mobiledocIsCompatibleWithV1(doc) {
+    if (doc
+        && doc.markups.length === 0
+        && doc.cards.length === 1
+        && doc.cards[0][0].match(/(?:card-)?markdown/)
+        && doc.sections.length === 1
+        && doc.sections[0].length === 2
+        && doc.sections[0][0] === 10
+        && doc.sections[0][1] === 0
+    ) {
+        return true;
+    }
+
+    return false;
+};
+
 module.exports.up = (options) => {
-    const postAllColumns = ['id', 'comment_id'];
+    const postAllColumns = ['id', 'comment_id', 'html', 'mobiledoc'];
 
     let localOptions = _.merge({
-        context: {internal: true}
+        context: {internal: true},
+        migrating: true
     }, options);
 
     common.logging.info(message1);
 
     return models.Post.findAll(_.merge({columns: postAllColumns}, localOptions))
         .then(function (posts) {
             return Promise.map(posts.models, function (post) {
-                if (post.get('comment_id')) {
-                    return Promise.resolve();
+                let mobiledoc;
+                let html;
+
+                try {
+                    mobiledoc = JSON.parse(post.get('mobiledoc') || null);
+                } catch (err) {
+                    common.logging.warn(`Invalid mobiledoc structure for ${post.id}. Falling back to blank structure.`);
+                    mobiledoc = converters.mobiledocConverter.blankStructure();
+                }
+
+                // CASE: convert all old editor posts to the new editor format
+                // CASE: if mobiledoc field is null, we auto set a blank structure in the model layer
+                // CASE: if html field is null, we auto generate the html in the model layer
+                if (mobiledoc && post.get('html') && post.get('html').match(/^<div class="kg-card-markdown">/)) {
```

Comment:
Is the "html field is null" case being handled? The conditional here doesn't seem to cover it but maybe that's handled at the model layer via `models.Post.edit()`?

Reply:
Hm i don't think so 🤔 I'll double check!

Reply:
Yeah good catch. If the html field is null, it wasn't regenerated. I'll push a commit 👍

Reply:
Pushed a commit.